provider/job_utils.py

```diff
@@ -71,12 +72,11 @@ def _wait_until_block_job_completed():
 
 @fail_on
 def block_job_complete(vm, job_id, timeout=120):
-    info = get_job_by_id(vm, job_id)
-    if info.get("type") == "mirror":
-        wait_until_job_status_match(vm, "ready", job_id, timeout)
-        arguments = {"device": job_id}
-        vm.monitor.cmd("block-job-complete", arguments)
-
+    wait_until_job_status_match(vm, "ready", job_id, timeout)
+    for event in vm.monitor.get_events():
+        if event.get("event") == BLOCK_JOB_READY_EVENT:
```

Comment:
what will happen if there are several jobs, we need to get the event of the job whose id is job_id, please refer to wait_until_block_job_completed

Reply:
have updated. please help to review again.

Reply:
IMO, we can keep your previous code change by now, and make a patch only if we hit some issues in future.
For the first version of job_utils.py, 'mirror' was checked because 
"This is supported for drive mirroring, where it also switches the device to write to the target path only"(from qmp doc)
In your testing, READY event is emitted because 
"If top == active, the job will not be completed by itself, user needs to complete the job with the block-job-complete command after getting the ready event" (from qmp doc)

But for the job_complete function itself, no matter the job type is 'mirror' or 'commit', we can always run 'job complete' command when and only when the status is READY(if we can make sure status changes to READY when ready event is emitted(my previous comment))

Reply:
have updated.